lib/galaxy/model/migrate/versions/0175_history_audit.py

```diff
@@ -0,0 +1,86 @@
+"""
+Add history audit table and associated triggers
+"""
+
+import datetime
+import logging
+
+from sqlalchemy import Column, DateTime, ForeignKey, Index, Integer, MetaData, Table
+
+from galaxy.model.migrate.triggers import (
+    history_update_time_field as old_triggers,  # rollback to old ones
+    update_audit_table as new_triggers,  # install me
+)
+from galaxy.model.migrate.versions.util import (
+    create_table,
+    drop_table
+)
+
+log = logging.getLogger(__name__)
+now = datetime.datetime.utcnow
+metadata = MetaData()
+
+# NOTE: A normal incrementing PK is not typically used in an audit table,
+# just foreign keys, but the ORM will probably choke without it
+AuditTable = Table(
+    "history_audit",
+    metadata,
+    Column("id", Integer, primary_key=True),
+    Column("history_id", Integer, ForeignKey("history.id"), nullable=False),
+    Column("update_time", DateTime, default=now, nullable=False),
+)
+
+Index('ix_history_audit_history_id_update_time_desc', AuditTable.c.history_id.desc(), AuditTable.c.update_time.desc())
```

Comment:
this won't be created in the database; you can move it (modifying the arguments) inside the AuditTable definition - then it will be created with the table when you call `create_table`. However,  you may not need it at all - see comment about composite PK (coming up)

Reply:
To be clear, it definitely does create the index in the database when referencing the table columns in this way. I can see it in my local DB right now. I chose this so that the max() based join would always choose the most recent row in the audit table. I'm open to other approaches though

<img width="391" alt="db-index" src="https://user-images.githubusercontent.com/290292/116921985-9508d800-ac09-11eb-8fc7-1d072f06f2fc.png">

Reply:
You are right - it does create it. (I'll comment on `max` performance below)